Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp

```diff
@@ -4311,17 +4309,14 @@ gboolean webkit_settings_apply_from_key_file(WebKitSettings* settings, GKeyFile*
         return FALSE;
     }
 
-    auto klass = G_OBJECT_GET_CLASS(settings);
-    unsigned totalProperties = 0;
-    GUniquePtr<GParamSpec*> properties(g_object_class_list_properties(klass, &totalProperties));
-
-    GRefPtr<GPtrArray> propertyNames = adoptGRef(g_ptr_array_sized_new(totalProperties));
-    GRefPtr<GArray> values = adoptGRef(g_array_sized_new(FALSE, FALSE, sizeof(GValue), totalProperties));
+    const auto properties = gObjectClassGetProperties(G_OBJECT_CLASS(settings));
```

Comment:
How about using `Vector<GParamSpec>`?

Reply:
Same here, `Vector<T>` cannot adopt a raw pointer. In this case there is a difference and it is that the returned container is heap-allocated, but the items are not: the `g_object_class_list_properties()` function is annotated as `(transfer container)`.